src/packet/bwe/mod.rs

```diff
@@ -66,19 +69,25 @@ impl SendSideBandwithEstimator {
     ) {
         let _ = self.started_at.get_or_insert(now);
 
-        let send_records: Vec<_> = records.collect();
-        let mut acked_packets = vec![];
+        let send_records: Vec<_> = records
+            .filter(|r| {
+                // Skip acked packets that have already been processed before.
+                !self.acked_packets_deduper.contains(r.seq())
+            })
+            .collect();
+        let mut acked_packets = Vec::with_capacity(send_records.len());
```

Comment:
Nice addition!

Reply:
A further optimization could be to reuse the allocation across calls

Reply:
Well, an obvious improvement would be removing that `let send_records = records.collect()` call. I think it can be done either by changing:

```diff
- LossController::update_bandwidth_estimate(packet_results: &[impl PacketResult])
+ LossController::update_bandwidth_estimate(packet_results: impl Iterator<Item = impl PacketResult>)
```
or by
```diff
- TwccSendRegister::send_records() -> Option<impl Iterator<Item = &TwccSendRecord>>
+ TwccSendRegister::send_records() -> Option<&[TwccSendRecord]>
```

I can look at it and maybe make another PR, if you would be ok with this.